### ISSUE TITLE
A child process process seems to survive .kill

### ISSUE THREAD

**Issue by zlotnleo** [crystal-lang/crystal#7047]
I am trying to create a child process and let it run for no longer than 2 seconds using the code below

```crystal
proc = Process.new("crystal", args: ["eval", "while true; end"])
d = delay(2.seconds) do
  proc.kill(Signal::KILL)
  puts "Killing the process!"
end
case proc.wait
when .normal_exit? then d.cancel
when .signal_exit? then puts("Timeout exceeded")
end
```

The output I'm getting is as expected:

```
Killing the process!
Timeout exceeded
```

While the code is running, `top` shows these 4 processes:
* `crystal`
* `crystal` (yes, another one)
* `crystal-run-mwe.tmp` (the filename is mwe.cr)
* `crystal-run-eval.tmp`

When the program finishes execution, all of them except for `crystal-run-eval.tmp` are gone. Since the only remaining process is the one containing an infinite loop, it's a bit of an issue.

Exactly the same behaviour is observed when instead of using `crystal eval`, I spawn a child process using `crystal run <filename>`, where filename is a crystal file with an infinite loop in it. 

Is it possible that `.kill` in the code actually kills the second `crystal` process instead of the `-eval.tmp` one?

Crystal version: Crystal 0.27.0
OS: macOS 10.14 (18A391)

**Comment by jhass** [comment]
My first guess: With a `SIGKILL` the `crystal eval` has no chance of forwarding that signal to its child, so it dies and the child gets reparented to its parent, which never waits on it or kills it either, so as it exits it gets reparented to its parent.

If you see the same behavior with a `SIGTERM` we likely miss to actually forward the signal to child, which might very well be, I didn't check.

**Comment by zlotnleo** [comment]
I used `SIGKILL` when I couldn't get the process to terminate from `.kill()`, which uses `SIGTERM` by default

**Comment by jhass** [comment]
So maybe `crystal eval` doesn't handle it at all, which might be.

**Comment by RX14** [comment]
`crystal eval` and `crystal run` just run the compiled binary as a child process and normal process semantics apply. When you `SIGTERM` or `SIGKILL` a process, it's children are not killed automatically. (when you press `ctrl-C` your shell sends a signal to the entire process group, so all processes stop)

The best solution here is to use process groups but I don't think crystal has support for them yet (unless you fork+exec manually). I'd suggest using `crystal build -o /tmp/myfile myfile.cr` then running `/tmp/myfile`. This will be exactly equivalent to `crystal run`.

This issue isn't a bug, but it would be nice to have a process group abstraction in the standard library.

**Comment by straight-shoota** [comment]
Closing per above comment. 
The feature for terminating the process group is tracked in #16616
